### PR TITLE
Feature: Drop support for Android 5 and 6 + Remove dead code

### PR DESCRIPTION
**Pull request by BradleyWire**
### Issues

Officially dropping support for Android 5 + 6 

### Solutions

Bump minSdkVersion to 24 (Android Nougat / 7.0) and remove the dead code associated with lower versions  

#### APK
[Download build #2383](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2383/artifact/build/artifact/wire-dev-PR2943-2383.apk)
[Download build #2390](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2390/artifact/build/artifact/wire-dev-PR2943-2390.apk)

### CODE REVIEW

**Comment by BradleyWire** [comment]
Still need to do some manual and automation regression before merging this in, but, it doesn't seem to introduce any obvious problems initially

**Comment by BradleyWire** [comment]
build again

**Comment by BradleyWire** [comment]
build again

**Comment by BradleyWire** [comment]
After doing some manual regression, it looks like the obvious features still work as intended.

File: app/src/main/scala/com/waz/zclient/utils/ContextUtils.scala [hunk 1]
```diff
@@ -83,13 +81,8 @@ object ContextUtils {
 
   def toPx(dp: Int)(implicit context: Context) = (dp * context.getResources.getDisplayMetrics.density).toInt
 
-  def getLocale(implicit context: Context) = {
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
-      DeprecationUtils.getDefaultLocale(context)
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/utils/ContextUtils.scala:88]: maybe we can clear this method from DeprecationUtils too?

File: app/src/main/java/com/waz/zclient/ui/colorpicker/ColorPickerDotView.java [hunk 2]
```diff
@@ -124,13 +112,7 @@ protected void onDraw(Canvas canvas) {
 
     private void drawSelectedRing(Canvas canvas) {
         int ringSize = getRingSize(dotRadius);
-        int whiteColor;
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-            //noinspection deprecation
-            whiteColor = getResources().getColor(R.color.draw_white);
-        } else {
-            whiteColor = getResources().getColor(R.color.draw_white, getContext().getTheme());
-        }
```
Comment (gizemb) [inline app/src/main/java/com/waz/zclient/ui/colorpicker/ColorPickerDotView.java:133]: there's actually a framework solution for this: `ContextCompat.getColor(context, colorRes)` which does exactly the same. we can update such usages with `ContextCompat`, `ResourcesCompat` etc in the future

Comment (BradleyWire) [reply app/src/main/java/com/waz/zclient/ui/colorpicker/ColorPickerDotView.java:133]: I was thinking the same thing, but, i didn't want to spend too much effort refactoring but concentrate on testing instead. Good point though 👍

**Comment by electrofloat** [comment]
So.. what does this mean for Android 5 & 6 users? Will we be able to use the android app in the future still? Or are we forced to buy a new phone by a messaging app?

**Pull request merged**